### PR TITLE
Fixed bug with incorrect Router Id 62.

### PR DESCRIPTION
**Pull request by rlubos**
This PR fixes issue #597.

### CODE REVIEW

File: src/core/thread/mle_router.hpp [hunk 1]
```diff
@@ -470,6 +470,17 @@ class MleRouter: public Mle
...
+    /**
+     * This method checks if a given Router ID has correct value.
+     *
+     * @param[in]  aRouterId  The Router ID value.
+     *
+     * @retval TRUE   If @p aRouterId is in correct range [0..62].
+     * @retval FALSE  If @p aRouterId is not a valid Router ID.
+     *
+     */
+    static bool IsRouterIdCorrect(uint8_t aRouterId) { return aRouterId <= kMaxRouterId; }
```
Comment (nibanks) [inline src/core/thread/mle_router.hpp]: Maybe `IsRouterIdValid` would be a better name for the function?

File: src/core/thread/mesh_forwarder.cpp [hunk 2]
```diff
@@ -515,7 +515,7 @@ ThreadError MeshForwarder::UpdateIp6Route(Message &aMessage)
             if (mMle.IsRoutingLocator(ip6Header.GetDestination()))
             {
                 rloc16 = HostSwap16(ip6Header.GetDestination().mFields.m16[7]);
-                VerifyOrExit(mMle.GetRouterId(rloc16) < Mle::kMaxRouterId, error = kThreadError_Drop);
+                VerifyOrExit(mMle.IsRouterIdCorrect(mMle.GetRouterId(rloc16)), error = kThreadError_Drop);
```
Comment (nibanks) [inline src/core/thread/mesh_forwarder.cpp]: Since it's a static function, you would only need to do `MleRouter::IsRouterIdCorrect`

Comment (darconeous) [inline src/core/thread/mesh_forwarder.cpp]: Eh, `mMle.` is a lot shorter than `MleRouter::`, and yields the exact same code. I'd keep it as-is myself.

Comment (rlubos) [inline src/core/thread/mesh_forwarder.cpp]: Well, reference to a static function through object seems to be used quite often in OpenThread, that's why I did it this way. Moreover, a reference to `Mle::` namespace would also be needed in this case so the function call will be even longer. I also opt to leave it as it is.

File: src/core/thread/mle_router.hpp (hunk 1 above)
Comment (rlubos) [inline src/core/thread/mle_router.hpp]: Agreed, done.

**Pull request merged**